helmfile.d/charts/prometheus-alerts/templates/alerts/opensearch.yaml

```
@@ -67,13 +67,13 @@ spec:
         summary: Opensearch Cluster Red (instance {{`{{ $labels.instance }}`}})
         description: Opensearch Cluster is in a Red status VALUE = {{`{{ $value }}`}}  LABELS = {{`{{ $labels }}`}}
     {{- range $prefixes := .Values.osIndexAlerts }}
-    - alert: {{ $prefixes.prefix | title }}SizeIncreasedOverLimit
-      expr: elasticsearch_indices_store_size_bytes_primary{namespace="opensearch-system",index=~"{{ $prefixes.prefix }}.+"} / (1024^2) > {{ $prefixes.alertSizeMB }}
+    - alert: OpenSearch{{ $prefixes.prefix | trimSuffix "-" | title }}IndexSizeOverLimit
+      expr: (elasticsearch_indices_store_size_bytes_primary{namespace="opensearch-system", index=~"{{ $prefixes.prefix }}.+"} / (1024 ^ 2) > {{ $prefixes.alertSizeMB }}) * clamp(rate(elasticsearch_indices_docs_primary{namespace="opensearch-system", index=~"{{ $prefixes.prefix }}.+"}[1h]) > 0, 1, 1)
```

```suggestion
      expr: (elasticsearch_indices_store_size_bytes_primary{namespace="opensearch-system", index=~"{{ $prefixes.prefix }}.+"} / (1024 ^ 2) > {{ $prefixes.alertSizeMB }}) * clamp(rate(elasticsearch_indices_docs_primary{namespace="opensearch-system", index=~"{{ $prefixes.prefix }}.+"}[2h]) > 0, 1, 1)
```
I took a look at long-running and found that increasing the sample window makes the metrics less choppy. But I still see that the active index can still be not considered as active with this function. So the alert can start flipping. 

But I do like this change and it will probably not happen that much. So I think it is fine to try it out like this and update it later on, if we notice that the alert is flipping between active and not.

---

> I took a look at long-running and found that increasing the sample window makes the metrics less choppy. But I still see that the active index can still be not considered as active with this function. So the alert can start flipping.

Sounds like we should spend some time to make sure that the alert don't start flipping.

---

I'm thinking, doesn't increasing the rate window make the alert fire for longer after it has been resolved, or capture the odd case where it is a long break in between new a new document being indexed? I have tried to look at the alert expression and I'm a bit puzzled on why it sometimes does not always consistently fire, I'm wondering if the exporter failed to scrape the metrics for some time.

---

@elastisys/goto-monitoring-stack do you have any input on this?

---

So I've had a look at this and here are my thoughts:

As @robinAwallace suggested 2h seems to be a good window, as there are frequent 1 hour windows where no documents are written to the `authlog` index, which would cause alerts to flip as the rate would go to 0 in this situation, and then back up once a document is written again.

And as @OlleLarsson pointed out, this would unfortunately have the side effect that an index would still considered active for 2 hours after a rollover. I think this is something that we can live with, and should be an improvement over the current state of these alerts.

---

> ..., I'm wondering if the exporter failed to scrape the metrics for some time.

My guess is timed out due to resource limits either on the exporter or OpenSearch itself.

> ..2h seems to be a good window, as there are frequent 1 hour windows where no documents are written to the `authlog` index, ..., this would unfortunately have the side effect that an index would still considered active for 2 hours after a rollover.

I think this is reasonable as it still would close eventually rather than when the index is deleted. I would be fine for it to be even longer if that would ensure it wouldn't flap.

---

I don't mind setting it to something like 4h either to be extra safe, does anyone object to this?